sdk/tables/data-tables/src/TablesBatch.ts

```diff
@@ -0,0 +1,321 @@
+ * @param credential credential to authenticate the batch request
+ */
+export function createBatch(
+  url: string,
+  tableName: string,
+  partitionKey: string,
+  credential?: TablesSharedKeyCredentialLike
+): TablesBatch {
+  const batchGuid = generateUuid();
+  const batchRequest = createInnerBatchRequest(batchGuid);
+
+  // Creates the pipeline that would intercept the requests
+  const pipeline = batchRequest.createPipeline();
+
+  // Client used to intercept the requests and add them to the batch instead of sending them to the service
+  const interceptClient: TableClient = new TableClient(url, tableName, {
+    requestPolicyFactories: pipeline
+  });
+
+  let batchUrl = url;
+
+  // Depending on the auth method used we need to build the url
+  if (!credential) {
+    // When authenticating with SAS we need to add the SAS token after $batch
+    const urlParts = url.split("?");
+    const baseUrl = urlParts[0];
+    const sas = urlParts.length > 1 ? `?${urlParts[1]}` : "";
+    batchUrl = `${baseUrl}/$batch${sas}`;
+  } else {
+    // When using a SharedKey credential no SAS token is needed
+    batchUrl = `${batchUrl}/$batch`;
+  }
+
+  return {
+    partitionKey,
+    async createEntity<T extends object>(entity: TableEntity<T>): Promise<void> {
+      if (entity.partitionKey !== this.partitionKey) {
+        throw new Error("All operations in a batch must target the same partitionKey");
+      }
+      await interceptClient.createEntity(entity);
+    },
+    async createEntities<T extends object>(entitites: TableEntity<T>[]): Promise<void> {
+      for (const entity of entitites) {
+        if (entity.partitionKey !== this.partitionKey) {
+          throw new Error("All operations in a batch must target the same partitionKey");
+        }
+        await interceptClient.createEntity(entity);
+      }
+    },
+    async deleteEntity(
+      partitionKey: string,
+      rowKey: string,
+      options?: DeleteTableEntityOptions
+    ): Promise<void> {
+      if (partitionKey !== this.partitionKey) {
+        throw new Error("All operations in a batch must target the same partitionKey");
+      }
+      interceptClient.deleteEntity(partitionKey, rowKey, options);
+    },
+    async updateEntity<T extends object>(
+      entity: TableEntity<T>,
+      mode: UpdateMode,
+      options?: UpdateTableEntityOptions
+    ): Promise<void> {
+      if (entity.partitionKey !== this.partitionKey) {
+        throw new Error("All operations in a batch must target the same partitionKey");
+      }
+      interceptClient.updateEntity(entity, mode, options);
+    },
+    async submitBatch(): Promise<any> {
+      const body = batchRequest.getHttpRequestBody();
+      const client = new ServiceClient();
+      const headers: RawHttpHeaders = {
+        accept: "application/json",
+        "x-ms-version": "2019-02-02",
+        "Accept-Charset": "UTF-8",
+        DataServiceVersion: "3.0;",
+        MaxDataServiceVersion: "3.0;NetFx",
```

Comment:
isn't NetFx a .NET thing?

Reply:
Suspect we need to do this because we are building the request ourselves (instead of allowing the generated client to do so).  Per https://docs.microsoft.com/en-us/rest/api/storageservices/setting-the-odata-data-service-version-headers it looks like the service expects.